### PR TITLE
Remove "unused" vertical-align declarations on Button & Button w/ Icon

### PR DESCRIPTION
**Pull request by KevinTCoughlin**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Chromium DevTools flags these usages of `vertical-align` as "unused" when capturing a "CSS Overview" snapshot on SharePoint Online and Fabric Storybook. I tested by manually modifying the styles and did not notice a visual difference. The fewer styles, the lighter the CSSOM.

Below screenshot is a capture of a SharePoint Online page.

image

Below screenshot is a capture of `master` Fabric Storybook.

image

Raising this PR to see if Screener fails visual regression test across the various permutations.

#### Focus areas to test

- Button and Icon visual regression tests should pass.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/12090)

cc: @patmill @antonlabunets

### CODE REVIEW

**Comment by KevinTCoughlin** [comment]
I'm unsure why `npm run update-snapshots` from root on this branch after fresh `yarn` and `yarn build`...

```shell
~\Work\office-ui-fabric-react [keco/button-vertical-align ≡]> npm run update-snapshots

> office-ui-fabric-react-repo@7.0.0 update-snapshots C:\Users\keco\Work\office-ui-fabric-react
> lerna run update-snapshots --stream

lerna notice cli v3.15.0
lerna info Executing command in 17 packages: "yarn run update-snapshots"
@uifabric/migration: yarn run v1.22.0
@uifabric/utilities: yarn run v1.22.0
@uifabric/migration: $ just-scripts jest -u
@uifabric/utilities: $ just-scripts jest -u
@uifabric/migration: [7:11:18 PM] ■ started 'jest'
@uifabric/migration: [7:11:18 PM] ■ Running Jest
@uifabric/migration: [7:11:18 PM] ■ C:\Users\keco\AppData\Local\nvs\node\12.14.0\x64\node.exe "C:\Users\keco\Work\office-ui-fabric-react\node_modules\jest\bin\jest.js" --config "C:\Users\keco\Work\office-ui-fabric-react\packages\migration\jest.config.js" --passWithNoTests --colors --updateSnapshot
@uifabric/utilities: [7:11:18 PM] ■ started 'jest'
@uifabric/utilities: [7:11:18 PM] ■ Running Jest
@uifabric/utilities: [7:11:18 PM] ■ C:\Users\keco\AppData\Local\nvs\node\12.14.0\x64\node.exe "C:\Users\keco\Work\office-ui-fabric-react\node_modules\jest\bin\jest.js" --config "C:\Users\keco\Work\office-ui-fabric-react\packages\utilities\jest.config.js" --passWithNoTests --colors --updateSnapshot
@uifabric/migration:  PASS  src/tests/cli.test.ts (5.18s)
@uifabric/migration:  PASS  src/tests/mods/7.0.0/warnAutobindRemoved.test.ts (6.031s)
@uifabric/migration:  PASS  src/tests/mods/7.0.0/searchBoxOnChangeProp.test.ts (6.042s)
@uifabric/migration:  PASS  src/tests/mods/7.0.0/createRefFromReact.test.ts (6.044s)
@uifabric/migration:  PASS  src/tests/mods/7.0.0/textfieldPropsChange.test.ts (6.058s)
@uifabric/migration:  PASS  src/tests/mods/7.0.0/comboboxPropsChange.test.ts (6.064s)
@uifabric/migration: [7:11:40 PM] ■ finished 'jest' in 21.83s
@uifabric/migration: Done in 24.25s.
@uifabric/utilities:  PASS  src/warn/warn.test.ts (7.965s)
@uifabric/utilities:  PASS  src/initials.test.ts (8.131s)
@uifabric/utilities:  PASS  src/dom/dom.test.ts (8.256s)
@uifabric/utilities:  PASS  src/math.test.ts (8.335s)
@uifabric/utilities:  PASS  src/initializeFocusRects.test.ts (8.424s)
@uifabric/utilities:  PASS  src/rtl.test.ts (8.46s)
@uifabric/utilities:  PASS  src/array.test.ts (8.471s)
@uifabric/utilities:  PASS  src/warn/warnControlledUsage.test.ts (8.57s)
@uifabric/utilities:  PASS  src/memoize.test.ts (8.636s)
@uifabric/utilities:  PASS  src/setFocusVisibility.test.ts
@uifabric/utilities:  PASS  src/selection/Selection.test.ts (8.726s)
@uifabric/utilities:  PASS  src/EventGroup.test.ts (8.738s)
@uifabric/utilities:  PASS  src/focus.test.tsx (8.947s)
@uifabric/utilities:  PASS  src/osDetector.test.ts
@uifabric/utilities:  PASS  src/mobileDetector.test.ts
@uifabric/utilities:  PASS  src/aria.test.ts
@uifabric/utilities:  PASS  src/object.test.ts
@uifabric/utilities:  PASS  src/classNamesFunction.test.ts
@uifabric/utilities:  PASS  src/customizations/Customizations.test.ts
@uifabric/utilities:  PASS  src/properties.test.ts
@uifabric/utilities:  PASS  src/customizations/customizable.test.tsx (9.555s)
@uifabric/utilities:  PASS  src/overflow.test.ts
@uifabric/utilities:  PASS  src/merge.test.ts
@uifabric/utilities:  PASS  src/componentAs/composeComponentAs.test.tsx
@uifabric/utilities:  PASS  src/renderFunction/composeRenderFunction.test.tsx
@uifabric/utilities:  PASS  src/customizations/Customizer.test.tsx (9.734s)
@uifabric/utilities:  PASS  src/styled.test.tsx (9.758s)
@uifabric/utilities:  PASS  src/appendFunction.test.ts
@uifabric/utilities:  PASS  src/controlled.test.ts
@uifabric/utilities:  PASS  src/keyboard.test.ts
@uifabric/utilities:  PASS  src/extendComponent.test.tsx
@uifabric/utilities:  PASS  src/css.test.ts
@uifabric/utilities:  PASS  src/initializeComponentRef.test.tsx
@uifabric/utilities:  PASS  src/BaseComponent.test.tsx
@uifabric/utilities:  PASS  src/asAsync.test.tsx
@uifabric/utilities:  PASS  src/safeSetTimeout.test.tsx
@uifabric/utilities:  PASS  src/safeRequestAnimationFrame.test.tsx
@uifabric/utilities: [7:11:52 PM] ■ finished 'jest' in 33.91s
@uifabric/utilities: Done in 36.33s.
@uifabric/styling: yarn run v1.22.0
@fluentui/react-theming: yarn run v1.22.0
@uifabric/styling: $ just-scripts jest -u
@fluentui/react-theming: $ just-scripts jest:snapshots
@fluentui/react-theming: [7:11:54 PM] x Unknown command: jest:snapshots
@uifabric/styling: [7:11:54 PM] ■ started 'jest'
@fluentui/react-theming: [7:11:54 PM] x Error previously detected. See above for error messages.
@uifabric/styling: [7:11:54 PM] ■ Running Jest
@uifabric/styling: [7:11:54 PM] ■ C:\Users\keco\AppData\Local\nvs\node\12.14.0\x64\node.exe "C:\Users\keco\Work\office-ui-fabric-react\node_modules\jest\bin\jest.js" --config "C:\Users\keco\Work\office-ui-fabric-react\packages\styling\jest.config.js" --passWithNoTests --colors --updateSnapshot
@fluentui/react-theming: error Command failed with exit code 1.
@fluentui/react-theming: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! yarn run update-snapshots exited 1 in '@fluentui/react-theming'
lerna WARN complete Waiting for 1 child process to exit. CTRL-C to exit immediately.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! office-ui-fabric-react-repo@7.0.0 update-snapshots: `lerna run update-snapshots --stream`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the office-ui-fabric-react-repo@7.0.0 update-snapshots script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\keco\AppData\Roaming\npm-cache\_logs\2020-02-27T03_11_54_886Z-debug.log
```

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.styles.ts [hunk 1]
```diff
@@ -51,7 +50,6 @@ export const getStyles = memoizeFunction(
           textDecoration: 'none',
           textAlign: 'center',
           cursor: 'pointer',
-          verticalAlign: 'top',
```
Comment (xugao) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.styles.ts]: hmm we have `display: 'inline-block',` set above.

Comment (xugao) [reply packages/office-ui-fabric-react/src/components/Button/BaseButton.styles.ts]: but i agree this should be removed :)

Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/Button/BaseButton.styles.ts]: Screener did not report any failures. Thus I suspect there is no delta in pixels painted by the browser with this attribute as it reported. If there is a regression, please escalate to me. Thanks.

**Pull request merged**